### ISSUE TITLE
DPE Inspector: A lot of properties across numerous components cannot be modified

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#16788]
**Describe the bug**
A lot of the components have one or more properties, that when modified revert back to their original value, refreshing the component as well. Property value type does not seem to matter, as it occurs on Number, Color, Vector3, Array, and slider properties. While we could not verify all of the components due to the time constraints, we have identified this issue on around 30 different components so far. 

For example here are the affected components presented in the attached video:
* *PostFX Layer* - Properties affected: *Layer Category* and *Priority in Default*.
* *Axis Aligned Box Shape* - Properties affected: All.
* *PhysX Dynamic Rigid Body* - Properties affected: *Type*, *Initial linear velocity*, *Initial angular velocity*, *Linear damping*, *Angular damping*, *Start asleep*, *Interpolate motion*, *Gravity enabled*.
* *Terrain Surface Materials List* - Properties affected: *Material Mappings* > element > *Material asset*.
* *Vegetation Layer Blender* - Properties affected: *Inherit Behavior*, *Propagate Behavior*, *Vegetation Areas* > element.

One constant between these resetting properties seems to be the Warnings printed to the *Console*:
```
[Warning] (Prefab Serialization) - The target path for "add" operation does not exist.

    Reason: The operation has halted during a merge operation for '/0' because a field or element has an invalid value.

[Warning] (Prefab) - Some of the patches were not successfully applied.
```

This issue occurs on both Windows and Linux.

**Steps to reproduce**
1. Create an entity and add one of the components mentioned above (for example *PostFX Layer*).
2. Change one of the mentioned properties (for example change *Layer Category* from *Default* to *FrontEnd*).

**Expected behavior**
Property is changed.

**Actual behavior**
Component is refreshed, property is reset to its original value.

**Video**

https://github.com/o3de/o3de/assets/86952082/7ac03749-477a-4487-8e4b-b6d2eaae796e

**Found in Branch**
Stabilization/2310 (d380ffa)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
d380ffaa3bdd7b4dad07f882acc856b64b209444

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 22.04.3 LTS
 - CPU AMD Ryzen 5 3600X 6-Core Processor
 - GPU AMD Radeon RX 5700 XT
 - Memory 16GB

**Comment by cgalvan** [comment]
This looks like it's from enabling prefab overrides.

**Comment by cgalvan** [comment]
When prefab overrides are enabled, we use the `AzToolsFramework::Prefab::PrefabComponentAdapter` instead of the normal `AZ::DocumentPropertyEditor::ComponentAdapter`. The prefab component adapter doesn't handle opaque values, and is likely missing some bug fixes that have gone into the DPE component adapater (since it hasn't been getting tested).

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2310 (811827a) build.
